### PR TITLE
close db when insert finished

### LINKED ISSUE #713: TelemetryService suspected memory leak.
**Describe the bug**

I try to use shifu to connect to five demo devices and use telemetry service to push data to tdengine with interval 1 second, After 1 days, I cannot to connect to my server, then I reboot my server, I check the prometheus in my cluster found the telemtryservice usage 5.2Gb
**To Reproduce**

1.deploy some mockdevice
2. use telemetry service to push data to tdengine
3. wait for some times

**Expected behavior**

With a normal memory usage

**Log Output**

nil

**Server (please complete the following information):**

 - Shifu Version: v0.26.0
 - Kubernetes distro k3s
 - Version v1.27.5+k3s1
 - CPU Architecture amd64

**Additional context**

image

### PR DESCRIPTION
**Pull request by leepala**
**What this PR does / why we need it**:
close db when insert finished
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #713

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [x] other (please specify)

**Special notes for your reviewer**:
close db when insert finished
**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/telemetryservice/sql/mysql/mysql.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (db *DBHelper) SendToDB(ctx context.Context, deviceName string, rawData []b
 		return err
 	}
 
-	return nil
+	return db.DB.Close()
```
Comment (tomqin93) [inline pkg/telemetryservice/sql/mysql/mysql.go:44]: `defer db.DB.Close()` if no error at line 37?

**Review by tomqin93 — COMMENTED** [review]
@leepala added 1 comment

File: pkg/telemetryservice/sql/mysql/mysql.go (hunk 1 above)
Comment (leepala) [reply pkg/telemetryservice/sql/mysql/mysql.go:44]: Defer does solve the problem, but it doesn't easily handle errors, and the direct return close method makes the code more concise.

Comment (leepala) [reply pkg/telemetryservice/sql/mysql/mysql.go:44]: Wait a minute I ask gpt

Comment (leepala) [reply pkg/telemetryservice/sql/mysql/mysql.go:44]: ok defer is better thanks for your comment tom😄

**Review by BtXin — APPROVED** [review]

**Review by tomqin93 — APPROVED** [review]
LGTM

**Pull request merged**